File: Arrays/03-Selection Sort In Array/selection-sort.cpp
#include <iostream>
using namespace std;

void ourSwap(int &x, int &y)
{
    int temp = x;
    x = y;
    y = temp;
}

int main()
{

    int arr[] = {10, 2, 3, 2, 1, 3, 4, 5, 6, 4, 3, 2, 1, 0};

    int n = sizeof(arr) / sizeof(arr[0]);

    for (int i = 0; i < n - 1; i++)
    {

        int index = i;
        for (int j = i + 1; j < n; j++)
        {

            if (arr[j] < arr[index])
            {
                index = j;
            }
        }

        ourSwap(arr[i], arr[index]); // passed by reference
    }

    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }

    return 0;
}

File: Arrays/03-Selection Sort In Array/selection-sort.js
let arr = [1, 3, 5, 3, 2, 3, 4, 5];

for (let i = 0; i < arr.length - 1; i++) {
  let index = i;

  for (let j = i + 1; j < arr.length; j++) {
    if (arr[j] < arr[index]) {
      index = j;
    }
  }

  swap(arr, i, index); // passed by value
}

function swap(arr, x, y) {
  let temp = arr[x];
  arr[x] = arr[y];
  arr[y] = temp;
}

console.log(arr);
